src/pages/spawn/spawnHost/spawnHostModal/PublicKeyForm.tsx

```diff
@@ -99,50 +104,47 @@ export const PublicKeyForm: React.FC<PublicKeyFormProps> = ({
                 })
               }
             />
-            {/* @ts-expect-error */}
             <StyledInput
               id="keyNameInput"
-              value={keyName}
+              aria-labelledby="Public Key Name"
               placeholder="Key name"
+              disabled={!savePublicKey}
+              spellCheck={false}
+              value={keyName}
               onChange={(e) => {
-                updatePublicKeyState({ key: publicKey, name: e.target.value });
+                updatePublicKeyState({
+                  key: publicKey,
+                  name: e.target.value,
+                });
               }}
-              disabled={!savePublicKey}
-              data-cy="key-name-input"
-              aria-label="Public Key Name"
             />
           </KeyNameContainer>
-        </FlexColumnContainer>
+        </>
       )}
     </Container>
   );
 };
 
-const FlexColumnContainer = styled.div`
+const Container = styled.div`
   display: flex;
   flex-direction: column;
+  width: 80%;
 `;
 
-const FlexContainer = styled.div`
+const KeyNameContainer = styled.div`
   display: flex;
-`;
-
-const KeyNameContainer = styled(FlexContainer)`
   align-items: center;
 `;
 
-const Container = styled(FlexColumnContainer)`
-  width: 80%;
+// @ts-expect-error
+const StyledRadioBoxGroup = styled(RadioBoxGroup)`
+  height: 48px;
+  margin: ${size.s} 0;
 `;
 
-const PaddedText = styled.span`
-  padding: 0 ${size.s};
-`;
-const SelectContainer = styled.div`
-  display: flex;
-  align-items: center;
-  justify-content: space-between;
-  flex-wrap: wrap;
+// @ts-expect-error
+const StyledSelect = styled(Select)`
+  z-index: ${zIndex.tooltip};
```

Comment:
Just a note: `popoverZIndex` on Select does NOT work for some reason so I had to use traditional CSS 🧍